server/services/ApplicationsDeveloper.js

```diff
@@ -144,18 +159,47 @@ class ApplicationsDeveloper {
       restrictedIPs: restrictedIps
     };
 
-    logger.info(`Service request to save stream key data against application=[${appId}], with payload=[${JSON.stringify(streamKeyPostRequest)}]`);
-    const streamKey = await client.streamKeysService.postStreamKey(streamKeyPostRequest, appId);
+    logger.info(
+      `Service request to save stream key data against application=[${appId}], with payload=[${JSON.stringify(
+        streamKeyPostRequest)}]`);
+    const streamKey = await client.streamKeysService.postStreamKey(
+      streamKeyPostRequest, appId);
 
     logger.debug(`streamKey=[${JSON.stringify(streamKey)}]`);
     return streamKey.resource;
   }
 
-  async updateKey (data, appId, keyId, oauthToken, environment) {
+  async updateKey (data, appId, keyId, keyType, oauthToken, environment) {
+    if (keyType === 'key') {
+      return this.updateRestApiKey(data, appId, keyId, oauthToken, environment);
+    } else if (keyType === 'stream-key') {
+      return this.updateStreamKey(data, appId, keyId, oauthToken, environment);
+    }
```

Comment:
If I tried to post a web or invalid key type would this ever resolve or would it get stuck in async hell

Reply:
I did have an else that would then return a new error like in add key but I took it out because I was worried about the merge conflicts with Robs branch, I will put it back in if it helps?

Reply:
I don't mind, I'll resolve the conflicts if you put it back in

Reply:
I have put a similar statement that was in the addKey method for the else to catch any invalid key types